circuit/list.go

```diff
@@ -0,0 +1,77 @@
+package circuit
+
+// a simple list to keep breakers sorted by access order.
+// It can return the first consecutive items that match a
+// condition (in our case, the ones that were inactive for a
+// while).
+type list struct {
```

Comment:
i also been waiting long for the use case ;)

i think it is simply optimization, but not premature, since it is already in the PR. Using an array would be like not using anything, because then we could also just iterate over the timestamps to see which breaker is idle. The reason for the optimization here is that this is in a synchronized scope, happening on every request, and blocking every concurrent request. Why do you think that it won't matter?